libraries/aws/shadow/src/aws_iot_shadow_parser.c

```diff
@@ -144,52 +141,53 @@ AwsIotShadowError_t _AwsIotShadow_ParseErrorDocument( const char * pErrorDocumen
                                    ERROR_DOCUMENT_CODE_KEY,
                                    ERROR_DOCUMENT_CODE_KEY_LENGTH,
                                    &pCode,
-                                   &codeLength ) == false )
-    {
-        /* Error parsing JSON document, or no "code" key was found. */
-        IotLogWarn( "Failed to parse code from error document.\n%.*s",
-                    errorDocumentLength,
-                    pErrorDocument );
-
-        IOT_SET_AND_GOTO_CLEANUP( AWS_IOT_SHADOW_BAD_RESPONSE );
-    }
-
-    /* Code must be in error document. */
-    AwsIotShadow_Assert( ( pCode > pErrorDocument ) &&
-                         ( pCode + codeLength < pErrorDocument + errorDocumentLength ) );
-
-    /* Convert the code to an unsigned integer value. */
-    code = ( uint32_t ) strtoul( pCode, NULL, 10 );
-
-    /* Parse the error message and print it. An error document must always contain
-     * a message. */
-    if( AwsIotDocParser_FindValue( pErrorDocument,
-                                   errorDocumentLength,
-                                   ERROR_DOCUMENT_MESSAGE_KEY,
-                                   ERROR_DOCUMENT_MESSAGE_KEY_LENGTH,
-                                   &pMessage,
-                                   &messageLength ) == true )
+                                   &codeLength ) == true )
     {
-        IotLogWarn( "Code %lu: %.*s.",
-                    code,
-                    messageLength,
-                    pMessage );
+        /* Code must be in error document. */
+        AwsIotShadow_Assert( ( pCode > pErrorDocument ) &&
+                             ( pCode + codeLength < pErrorDocument + errorDocumentLength ) );
+
+        /* Convert the code to an unsigned integer value. */
+        code = ( uint32_t ) strtoul( pCode, NULL, 10 );
```

Comment:
Not sure if the string to integer conversion is actually required here, as it seems like the error code will only be logged, which can be done with the string version stored in `pCode`

Reply:
That code is also used here:
```
status = _codeToShadowStatus( code );
```